configuration/configurator/ConfigNotifications.cpp

```diff
@@ -0,0 +1,106 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2015 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "SchemaCommon.hpp"
+#include "ConfigNotifications.hpp"
+#include "jlib.hpp"
+#include "jhash.ipp"
+#include "jhash.hpp"
+#include "jstring.hpp"
+
+using namespace CONFIGURATOR;
+
+CNotificationManager* CNotificationManager::s_pNotificationManager = NULL;
+
+CNotificationManager::CNotificationManager()
+{
+}
+
+CNotificationManager *CNotificationManager::getInstance()
+{
+    if (s_pNotificationManager == NULL)
```

Comment:
As part of work to support multi-threading.  At this point the code is single threaded/single session only.